scripts/cfos/README.md

```diff
@@ -50,10 +53,110 @@
 
 #### **validate.py**
 ##### Description
-    WIP: Non-executable python file that contains the validation rules for each column in the input excel file. The validation rules are applied using pandas_schema.
+    Dynamically build validation using pandas_schema. Validation is built table by table (tables as defined in the schema section of the schema.json file). 
+
+	Dynamic validation is built based on the field_definitions section of the schema.JSON file that is provided alongside this script. 
+
+	What validation is applied to what table is determined based on the schema definitions in the schema.JSON file. 
+
+	See the "SCHEMA SETUP" section below for how exactly the schema.JSON file is used for this.
 
 #### **dataset_tables_schema.json**
-##### Description
-    WIP: File containing a dataset (by name) and the dataset's tables, as well as each table's organization structure - which columns are organized into which tables for a given dataset.
+#### Schema Setup
+
+This is a JSON file used to set up validation for expected columns, values, and input schemas for data being loaded into Terra. 
+
+The JSON is split into two sections. One section lists and describes all the columns of data expected in the input. By default, these columns are treated as required for successful data ingest. Another section defines various schemas which indicate the expected organization of the files, both in the input file and upon upload to the database.
+
+What are schemas? 
+Schemas are different ways of organizing the columns of data. This will determine how many tables will be created to store the data in Terra, and which columns appear in which tables. All columns specified in the schema being used must be present in the input file for successful data upload.
+
+Which schema to use is specified as an input to the upload script at run time.
+
+How do you define columns?
+ Columns of data are defined in the “Fields” section of the JSON document. Each column must have a unique column_ID, which is used to identify the column in the input document. The column_ID is the value under which the rest of the column information is stored in the JSON dictionary. 
+
+Note: if you have the same field in multiple tables, the same validation will apply to both, unless they are named something unique. AKA: if "filename" in one table should be alphanumeric and in another table should be an integer, you need to name the fields "tablex_filename" and tabley_filename" to apply different validation.
+
+
+Each field has the following attributes. Attributes that are “type-specific” are listed under the corresponding field “data type”. 
+
+
+Field:
+column_ID
+field_name
+field_type
+value_required (optional - defaults to “yes?”)
```

Comment:
Good catch - pretty sure this is because I made this value call but wanted my reviewer to confirm that's what they would expect as well. Then I forgot about it. :) 

Does this match your expectations? I'll go ahead and remove the "?" since it's served its purpose.

Reply:
I actually had this as "defaults to no" in the code, so I updated this to be consistent with it for now, but can change both if we think it should be the opposite.